PWGHF/D2H/Tasks/taskDplus.cxx

```diff
@@ -167,25 +247,80 @@ struct HfTaskDplus {
 
   // Fill THnSparses for the ML analysis
   /// \param candidate is a particle candidate
+  /// \param ptbhad transverse momentum of beauty mother for nonprompt candidates
+  /// \param flagBHad transverse momentum of beauty mother for nonprompt candidates
+  /// \param centrality collision centrality
+  /// \param occupancy collision occupancy
   template <bool isMc, bool isMatched, typename T1>
-  void fillSparseML(const T1& candidate)
+  void fillSparseML(const T1 candidate,
```

Comment:
Don't remove the reference from `candidate`.

Reply:
Why should I keep it in the candidate and remove from the other variables? We don't want to modify it in the fill function

Reply:
The reference is useless for small base types (like `float`, `int`) which are cheap to copy. It is needed in case of `candidate` because it's a bigger type. Using `const` disables the possibility to modify the referenced value inside the function.